### PR TITLE
Makes PropertyGrid for CoboBox show "(none)" when DataSource is null …

### LINKED ISSUE #12570: No "(none)" text display for the DataSource property in PropertyGrid control
### .NET version

.NET SDK 10.0.100-alpha.1.24573.1

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

The DataSource property was enabled since .NET 9.0

### Issue description

No "(none)" text display for the DataSource property in PropertyGrid control

**.NET :**
Image

**.NET Farmework:**
Image

### Steps to reproduce

1. Create a .NET WinForms project
2. Add a PropertyGrid & ComboBox control on the Form1.cs[Design]
3. Select the comboBox1 as Object for propertyGrid1
4. Observe the DataSource property value

### PR DESCRIPTION
**Pull request by ricardobossan**
Fixes #12570

## Root Cause

The `ComboBox.DataSource` property was missing a `TypeConverter` to handle null value display in `PropertyGrid`. When `DataSource` is null, `PropertyGrid` shows empty/blank instead of "(none)" text, breaking consistency with .NET Framework behavior and other similar properties like `DisplayMember` and `ValueMember`.

## Proposed changes

- Created `DataSourceConverter` class that inherits from `ReferenceConverter` and converts null values to "(none)" for string display
- Applied `TypeConverter` attribute to `ComboBox.DataSource` property to use the new converter
- Added comprehensive unit tests to verify converter behavior across different cultures and scenarios

## Customer Impact

Restores .NET Framework parity for PropertyGrid display. Developers using `PropertyGrid` to inspect `ComboBox` properties will now see consistent "(none)" text when `DataSource` is null, improving design-time experience.

## Regression?

No. This is a missing feature/behavior that existed in .NET Framework but was lost in .NET Core migration.

## Risk

Minimal.

### Before
before

### After  
after

## Test methodology

- Unit tests for `DataSourceConverter` covering null conversion, culture handling, and edge cases
- Manual testing with `PropertyGrid` and `ComboBox` to verify "(none)" display

## Test environment(s)

- 10.0.100-alpha.1.24573.1

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/13860)

### CODE REVIEW

File: src/System.Windows.Forms/System/Windows/Forms/Controls/ComboBox/ComboBox.cs [hunk 1]
```diff
@@ -405,6 +405,7 @@ protected override Size DefaultSize
     [RefreshProperties(RefreshProperties.Repaint),
     AttributeProvider(typeof(IListSource))]
     [SRDescription(nameof(SR.ListControlDataSourceDescr))]
+    [TypeConverter("System.Windows.Forms.Design.DataSourceConverter, System.Windows.Forms.Design")]
```
Comment (Epica3055) [inline src/System.Windows.Forms/System/Windows/Forms/Controls/ComboBox/ComboBox.cs]: why put DataSourceConverter in Design project?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataSourceConverter.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public DataSourceConverter() : base(typeof(IListSource))
+    {
+    }
+
+    public override object? ConvertTo(ITypeDescriptorContext? context, CultureInfo? culture, object? value, Type destinationType)
+    {
+        if (destinationType == typeof(string) && value is null)
+        {
+            return SR.None_lc;
+        }
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataSourceConverter.cs:20]: LGTM! However, according to [the logic in .NetFramework](https://devdiv.visualstudio.com/DevDiv/_git/DotNetFramework?path=/src/ndp/fx/src/Designer/WinForms/System/WinForms/Design/DataSourceConverter.cs&version=GBmain&line=101&lineEnd=104&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), any value here should be converted to string type

File: src/System.Windows.Forms/System/Windows/Forms/Controls/ComboBox/ComboBox.cs (hunk 1 above)
Comment (ricardobossan) [reply src/System.Windows.Forms/System/Windows/Forms/Controls/ComboBox/ComboBox.cs]: DataSourceConverter is for PropertyGrid display at design-time only. It follows the same pattern as other PropertyGrid converters like DataMemberFieldConverter which are also in the Design project.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataSourceConverter.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Globalization;
+
+namespace System.Windows.Forms.Design;
+
+internal class DataSourceConverter : ReferenceConverter
```
Comment (KlausLoeffelmann) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataSourceConverter.cs:9]: If have a few questions here:

* What was/is the behavior in NetFX here? Do we also have such a converter there?
* Is this just for the ComboBox? How does the ListBox for example handle this?

Comment (ricardobossan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataSourceConverter.cs:9]: - Yes, `DataSourceConverter` [also exists in NetFX](https://vstfdevdiv.corp.microsoft.com/DevDiv2/DevDiv/_versionControl?path=%24/DevDiv/FX/Feature/NetFXDev1/NDP/fx/src/Designer/WinForms/System/WinForms/Design/DataSourceConverter.cs&version=T&line=16&lineEnd=17&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).

- You're right! I should have added `TypeConverter` to the base `ListControl.DataSource` property instead of just the `ComboBox`. That way both inheriting controls get the same behavior.

I have updated the code and this is how it looks after the latest fix:

13860_LISTBOX_FIXED

Thank you for the guidance!
